tools/tabix_index.cwl

```diff
@@ -0,0 +1,34 @@
+cwlVersion: v1.0
+class: CommandLineTool
+id: tabix_index 
+doc: >-
+  This tool will run tabix conditionally dependent on whether an index is provided.
+  The tool will output the input_file with the index, provided or created within, as a secondary file.
+requirements:
+  - class: ShellCommandRequirement
+  - class: DockerRequirement
+    dockerPull: 'kfdrc/samtools:1.9'
+  - class: InlineJavascriptRequirement
+  - class: ResourceRequirement
+    coresMin: 16
```

Comment:
16 cores seems a little aggressive for indexing. Also, I've found that for bgzip, multithreading is definitely leveraged, for indexing, not so much 😕